core/src/main/java/io/parsingdata/metal/expression/comparison/Eq.java

```diff
@@ -40,9 +40,13 @@ public Eq(final ValueExpression value, final ValueExpression predicate) {
     public boolean compare(final Value left, final Value right) {
         final byte[] leftBytes = left.getValue();
```

Comment:
You could  perhaps also now pre-check on slice size, so that when these do not match, no data is read?